server/services/prisonService.test.ts

```diff
@@ -0,0 +1,211 @@
+import type { PrisonResponse } from 'prisonRegisterApiClient'
+import toPrison from '../data/mappers/prisonMapper'
+import PrisonService from './prisonService'
+import PrisonRegisterStore from '../data/cache/prisonRegisterStore'
+import PrisonRegisterClient from '../data/prisonRegisterClient'
+import aValidPrisonResponse from '../testsupport/prisonResponseTestDataBuilder'
+import aValidPrison from '../testsupport/prisonTestDataBuilder'
+
+jest.mock('../data/mappers/prisonMapper')
+
+describe('prisonService', () => {
+  const mockedPrisonMapper = toPrison as jest.MockedFunction<typeof toPrison>
+
+  const prisonRegisterStore = {
+    getActivePrisons: jest.fn(),
+    setActivePrisons: jest.fn(),
+  }
+
+  const prisonRegisterClient = {
+    getAllPrisons: jest.fn(),
+  }
+
+  const prisonService = new PrisonService(
+    prisonRegisterStore as unknown as PrisonRegisterStore,
+    prisonRegisterClient as unknown as PrisonRegisterClient,
+  )
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  const allPrisons: Array<PrisonResponse> = [
+    aValidPrisonResponse({
+      prisonId: 'ACI',
+      prisonName: 'Altcourse (HMP)',
+      active: false,
+    }),
+    aValidPrisonResponse({
+      prisonId: 'ASI',
+      prisonName: 'Ashfield (HMP)',
+      active: true,
+    }),
+    aValidPrisonResponse({
+      prisonId: 'MDI',
+      prisonName: 'Moorland (HMP & YOI)',
+      active: true,
+    }),
+  ]
+
+  const activePrisons: Array<PrisonResponse> = [
```

Comment:
I'm a bit confused between the terminology of `active` and `cached`. Are they not the same thing?

Reply:
No, defo not the same thing.
"cached" is a tech term, and is what you imagine it is (ie. an object in the Redis cache and therefore we dont need to look it up from it's source REST API)

"active" in this sense is a HMPPS term. Prisons have a field `active` which is a boolean and determines whether the prison is active of not. IE. some prisons are non-active for some reason (perhaps they have been closed). These prisons come back from the prison-register-api REST API call, but are not in use and have no prisoners in them.
These prisons we dont care about, so we dont cache them.